### ISSUE TITLE
Tensorboard callback not logging learning rate when not using LearningRateSchedule (e.g. when using ReduceLROnPlateau)

### ISSUE THREAD

**Issue by MatejHl** [tensorflow/tensorflow#46248]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Linux Ubuntu 18.04
- TensorFlow installed from: pip3
- TensorFlow version: tf-nightly 2.5.0-dev20210104
- Python version: 3.8.5
- CUDA/cuDNN version: 11.0.3 / 8.0.4
- GPU model and memory: Quadro RTX 5000 / 16384 MB

**Describe the current behavior**  
If there is no LearningRateSchedule (either LR is constant or is changed via callback) then [Tensorboard callback](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/TensorBoard) doesn't log learning rate. This is especially relevant when using [ReduceLROnPlateau](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/ReduceLROnPlateau) callback.

**Describe the expected behavior**  
Learning rate should be always logged or at least in the case of using a callback.

**Standalone code to reproduce the issue**  
```
import tensorflow as tf
import datetime
import os

class SimpleModel(tf.keras.Model):
    def __init__(self):
        super(SimpleModel, self).__init__()
        self.Dense_in = tf.keras.layers.Dense(units = 50, 
                                activation=tf.nn.relu, 
                                use_bias=True)
        self.Dense_hidden = tf.keras.layers.Dense(units = 20, 
                                activation=tf.nn.relu, 
                                use_bias=True)
        self.Dense_out = tf.keras.layers.Dense(units = 1, 
                                activation=None, 
                                use_bias=True)

    def call(self, inputs, training = True):
        X = self.Dense_in(inputs)
        X = self.Dense_hidden(X)
        X = self.Dense_out(X)
        return X

dummy_data = tf.random.normal((10000, 50))
dummy_y = tf.random.uniform((10000,), minval=0, maxval=2, dtype=tf.dtypes.int32)


#### LR is not logged here ####
model = SimpleModel()

loss = tf.keras.losses.BinaryCrossentropy(name='binary_crossentropy', from_logits=True)
opt = tf.keras.optimizers.Adam(learning_rate = 0.001)
model.compile(optimizer = opt, loss = loss, metrics=["acc"])

_datetime = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
log_dir = os.path.join('dummy_logs', _datetime)
tb_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir)

model.fit(x = dummy_data, y = dummy_y, epochs = 10,
        callbacks=[tf.keras.callbacks.ReduceLROnPlateau(monitor='acc', 
                        factor=0.5, patience=10, verbose=0, mode='auto', 
                        min_delta=0.0001, cooldown=0, min_lr=0),
                    tb_callback])

#### LR is logged here ####
model = SimpleModel()

lr_schedule = tf.keras.optimizers.schedules.ExponentialDecay(
    initial_learning_rate = 0.001,
    decay_steps=100000,
    decay_rate=0.96)

loss = tf.keras.losses.BinaryCrossentropy(name='binary_crossentropy', from_logits=True)
opt = tf.keras.optimizers.Adam(learning_rate = lr_schedule)
model.compile(optimizer = opt, loss = loss, metrics=["acc"])

_datetime = datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
log_dir = os.path.join('dummy_logs', _datetime)
tb_callback = tf.keras.callbacks.TensorBoard(log_dir=log_dir)

model.fit(x = dummy_data, y = dummy_y, epochs = 10,
        callbacks=[tb_callback])
```
![Screenshot from 2021-01-07 12-14-15](https://user-images.githubusercontent.com/32313554/103887434-517ee100-50e3-11eb-9597-4978bee27972.png)

**Other info / logs**   
I think that the problem is in tf.keras.callbacks.TensorBoard in _collect_learning_rate (see code block below). If self.model.optimizer.lr is sheduler then logs['learning_rate'] is created and LR can be seen in tensorboard. However if self.model.optimizer.lr is not an instance of learning_rate_schedule.LearningRateSchedule then logs['learning_rate'] does not exists and thus LR is not logged.
```
def _collect_learning_rate(self, logs):
    lr_schedule = getattr(self.model.optimizer, 'lr', None)
    if isinstance(lr_schedule, learning_rate_schedule.LearningRateSchedule):
      logs['learning_rate'] = lr_schedule(self.model.optimizer.iterations)
    return logs
```

A quick fix that I see here could be to add *else* or *elif* and directly log self.model.optimizer.lr. Changing as below fixed the issue for me when using ReduceLROnPlateau callback. This fix assumes that if no LearningRateSchedule is used then self.model.optimizer.lr could be logged as scalar by tensorboard callback. If this is not always possible then I guess adding some reasonable *elif* to be able to catch LR that can be logged (e.g. scalar) could resolve the problem.
```
class TensorBoardCallback(tf.keras.callbacks.TensorBoard):
    def __init__(self, **kwargs):
        super(TensorBoardCallback, self).__init__(**kwargs)

    def _collect_learning_rate(self, logs):
        lr_schedule = getattr(self.model.optimizer, 'lr', None)
        if isinstance(lr_schedule, tf.keras.optimizers.schedules.LearningRateSchedule):
            logs['learning_rate'] = lr_schedule(self.model.optimizer.iterations)
        elif lr_schedule is not None:
            logs['learning_rate'] = lr_schedule
        return logs
```

**Comment by amahendrakar** [comment]
@MatejHl,
Tensorboard issues are tracked in tensorflow/tensorboard repo. Could you please submit a new issue from [this link](https://github.com/tensorflow/tensorboard/issues/new/choose) and fill in the template, so that we can track the issue there. Thanks!

**Comment by MatejHl** [comment]
Thanks for fast response. 
I opened new issue [here](https://github.com/tensorflow/tensorboard/issues/4522).
